packages/flame_3d/lib/src/resources/geometry/geometry.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:typed_data';
+
+import 'package:flame_3d/resources.dart';
+import 'package:flutter_gpu/gpu.dart' as gpu;
+
+class Geometry extends Resource<gpu.DeviceBuffer?> {
+  Geometry({List<Vertex>? vertices, List<int>? indices}) : super(null) {
+    if (vertices != null) {
+      setVertices(vertices);
+    }
+    if (indices != null) {
+      setIndices(indices);
+    }
+  }
+
+  ByteBuffer _vertices = ByteData(0).buffer;
+  int _vertexCount = 0;
+
+  ByteBuffer _indices = ByteData(0).buffer;
+  int _indexCount = 0;
+
+  @override
+  gpu.DeviceBuffer? get resource {
+    var resource = super.resource;
+    final sizeInBytes = _vertices.lengthInBytes + _indices.lengthInBytes;
+    if (resource?.sizeInBytes != sizeInBytes) {
+      resource = super.resource = gpu.gpuContext.createDeviceBuffer(
+        gpu.StorageMode.hostVisible,
+        sizeInBytes,
+      );
+
+      resource!.overwrite(_vertices.asByteData());
+
+      resource.overwrite(
+        _indices.asByteData(),
+        destinationOffsetInBytes: _vertices.lengthInBytes,
+      );
+    }
+    return resource;
+  }
+
+  // TODO(bdero): This should have an attribute map instead and be fully SoA,
```

Comment:
Haha, we have comments from brandon in here, that's great :smile:

Reply:
I copied those in from the examples I used from him, because they are 100% valid comments 😂